Remotion/Validation/Core/Validators/AbstractValueComparisonValidator.cs

```diff
@@ -13,33 +25,60 @@ namespace Remotion.Validation.Validators
   public abstract class AbstractValueComparisonValidator : PropertyValidator, IValueComparisonValidator
   {
     public object ValueToCompare { get; }
+    public abstract Comparison Comparison { get; }
+    private IComparer Comparer { get; }
 
-    protected AbstractValueComparisonValidator (IComparable value, string errorMessage, IValidationMessage validationMessage)
+    protected AbstractValueComparisonValidator ([NotNull] IComparable value, [CanBeNull] IComparer comparer, [NotNull] string errorMessage, [NotNull] IValidationMessage validationMessage)
         : base (errorMessage, validationMessage)
     {
       ArgumentUtility.CheckNotNull ("value", value);
 
+      Comparer = comparer;
       ValueToCompare = value;
-
-      //TODO RM-5906: Add System.IComparer, similar to NotEqualValidator with System.IEqualsComparer
-      //TODO RM-5906: Refactor to do comparison based on enum and use derived types only to indicate type etc.
     }
 
-    public abstract Comparison Comparison { get; }
-
-    protected abstract bool IsValid (IComparable value, IComparable valueToCompare);
-
     protected sealed override bool IsValid (PropertyValidatorContext context)
     {
-      if (context.PropertyValue == null)
+      var propertyValue = context.PropertyValue;
+      if (Comparer != null)
```

Comment:
interesting choice, doing this before testing the property value for null. Hmm...I think we should test first for null and only then use the Comparer.

also, possibly it would be cleaner to do just one IsComparsonResultValid. not 100% sure, though.